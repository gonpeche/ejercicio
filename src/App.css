.App {
  text-align: center;
  margin: 1%;
  display: flex;
  flex-direction: row;
  margin-left: 10%;
}

.contenido {
  border: 2px solid black;
  padding: 10px;
  width: 100%;
  margin-top: 1px;
  margin-left: 50%;
}

.notas {
  width: 50%;
  margin-left:10%;
}
