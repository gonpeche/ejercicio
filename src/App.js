import React from 'react';
import './App.css';
import CargaProducto from './CargaProducto';
import Venta from './Venta';
import Compra from './Compra';
import CambioDeposito from './CambioDeposito';
import Consolidado from './Consolidado';

import { updateStock, init } from './utils';
class App extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
        consolidado: '',
        existencias: ''
    }
    this.update = this.update.bind(this)
  }

  componentDidMount() {
    const consolidado = localStorage.getItem('consolidado');
    const existencias = localStorage.getItem('existencias');
    this.setState({ consolidado, existencias })
    init()
  }

  update(sku, deposito, cantidad, tipo) {
    updateStock(sku, deposito, cantidad, tipo);
    this.setState({
      consolidado: localStorage.getItem('consolidado'),
      existencias: localStorage.getItem('existencias')
    })
  }

  render() {
    const { consolidado } = this.state
    return (
      <div className="App">
        <div className="sidebar">
          <CargaProducto />
          <Venta update={this.update}/>
          <Compra update={this.update} />
          <CambioDeposito />        
        </div>
        <div className="consolidado">
          <Consolidado 
            consolidado={consolidado ? consolidado : null}
          />
        </div>
      </div>
    );
  }
}

export default App;
