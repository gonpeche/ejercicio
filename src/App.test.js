import React from 'react';
import App from './App';
import renderer from 'react-test-renderer'

describe('Main component', () => {
  it('renderea Ok', () => {
    const tree = renderer
      .create(<App />)
      .toJSON();
      expect(tree).toMatchSnapshot();
  })
})


