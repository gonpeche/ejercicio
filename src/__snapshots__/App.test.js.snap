// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main component renderea Ok 1`] = `
<div
  className="App"
>
  <div
    className="sidebar"
  >
    <div
      style={
        Object {
          "border": "2px solid black",
          "padding": "10px",
          "width": "300px",
        }
      }
    >
      CARGA PRODUCTO:
      <form
        onSubmit={[Function]}
      >
        <div>
          <div>
            <label>
              Empresa:
              <input
                name="empresa"
                onChange={[Function]}
                type="text"
                value=""
              />
            </label>
          </div>
          <div>
            <label>
              Marca:
              <input
                name="marca"
                onChange={[Function]}
                type="text"
                value=""
              />
            </label>
          </div>
          <div>
            <label>
              SKU:
              <input
                name="SKU"
                onChange={[Function]}
                type="number"
                value=""
              />
            </label>
          </div>
          <div>
            <label>
              Nombre Comercial:
              <input
                name="nombreComercial"
                onChange={[Function]}
                type="text"
                value=""
              />
            </label>
          </div>
          <div>
            <label>
              Volumen:
              <input
                name="volumen"
                onChange={[Function]}
                type="text"
                value=""
              />
            </label>
          </div>
          <div>
            <label>
              Descripcion:
              <input
                name="descripcion"
                onChange={[Function]}
                type="text"
                value=""
              />
            </label>
          </div>
          <input
            type="submit"
            value="Submit"
          />
        </div>
      </form>
    </div>
    <div
      style={
        Object {
          "border": "2px solid black",
          "marginTop": "2px",
          "padding": "10px",
          "width": "300px",
        }
      }
    >
      VENTA:
      <form
        onSubmit={[Function]}
      >
        <div>
          <div>
            <label>
              SKU:
              <input
                name="SKU"
                onChange={[Function]}
                type="number"
                value=""
              />
            </label>
          </div>
          <div>
            <label>
              Cantidad:
              <input
                name="cantidad"
                onChange={[Function]}
                type="text"
                value=""
              />
            </label>
          </div>
          <div>
            <label>
              Deposito:
              <input
                name="deposito"
                onChange={[Function]}
                type="text"
                value=""
              />
            </label>
          </div>
          <div>
            <label>
              Precio Venta:
              <input
                name="ventas"
                onChange={[Function]}
                type="number"
                value=""
              />
            </label>
          </div>
          <div>
            <label>
              Mes:
              <input
                name="mes"
                onChange={[Function]}
                type="text"
                value=""
              />
            </label>
          </div>
          <input
            type="submit"
            value="Submit"
          />
        </div>
      </form>
    </div>
    <div
      style={
        Object {
          "border": "2px solid black",
          "marginTop": "2px",
          "padding": "10px",
          "width": "300px",
        }
      }
    >
      COMPRA:
      <form
        onSubmit={[Function]}
      >
        <div>
          <div>
            <label>
              SKU:
              <input
                name="SKU"
                onChange={[Function]}
                type="number"
                value=""
              />
            </label>
          </div>
          <div>
            <label>
              Cantidad:
              <input
                name="cantidad"
                onChange={[Function]}
                type="text"
                value=""
              />
            </label>
          </div>
          <div>
            <label>
              Deposito:
              <input
                name="deposito"
                onChange={[Function]}
                type="text"
                value=""
              />
            </label>
          </div>
          <div>
            <label>
              Costo Compra:
              <input
                name="compras"
                onChange={[Function]}
                type="number"
                value=""
              />
            </label>
          </div>
          <div>
            <label>
              Mes:
              <input
                name="mes"
                onChange={[Function]}
                type="text"
                value=""
              />
            </label>
          </div>
          <input
            type="submit"
            value="Submit"
          />
        </div>
      </form>
    </div>
  </div>
  <div
    className="consolidado"
  />
</div>
`;
